### PR TITLE
Capture errors in PostHog

### PR DESCRIPTION
**Pull request by nop33**
@mvaivre I purposefully did not capture the actual error message in the spirit of preserving user privacy. We don't know what info the error messages might include. Some could be PII. To respect our decision to not store any user data, I simply capture generic errors that will give us a hint as to where there might be a problem so that we can identify it outselves.

I did a global search for `catch (e) {` and capture an error.

### CODE REVIEW

File: src/components/PasswordConfirmation.tsx [hunk 1]
```diff
@@ -61,6 +61,7 @@ const PasswordConfirmation: FC<PasswordConfirmationProps> = ({
         onCorrectPasswordEntered(password)
       }
     } catch (e) {
+      // Discuss: Do we want to capture password validation errors?
```
Comment (nop33) [inline src/components/PasswordConfirmation.tsx:64]: I would again say no.

File: src/contexts/global.tsx [hunk 2]
```diff
@@ -90,6 +90,7 @@ export const GlobalContextProvider: FC<{ overrideContextValue?: PartialDeep<Glob
       wallet = WalletStorage.load(walletId, password)
     } catch (e) {
       console.error(e)
+      // Discuss: Do we want to capture password validation errors?
```
Comment (nop33) [inline src/contexts/global.tsx:93]: Same here

File: src/contexts/walletconnect.tsx [hunk 3]
```diff
@@ -313,10 +317,13 @@ export const WalletConnectContextProvider: FC = ({ children }) => {
 
         if (!errorMessage.includes('Pairing already exists')) {
           dispatch(walletConnectPairingFailed(errorMessage))
+          posthog.capture('Error - Could not pair with WalletConnect')
+        } else {
+          posthog.capture('Error - Could not pair with WalletConnect: Pairing already exists')
```
Comment (nop33) [inline src/contexts/walletconnect.tsx:322]: An issue that I see with this is that we'll spam PostHog a bit... I am not sure if it's our implementation problem or a WC problem, but I very often get these errors in the console, without affecting the normal usage of the wallet.

File: src/App.tsx [hunk 4]
```diff
@@ -129,6 +131,7 @@ const App = () => {
       // TODO: Check if connection to explorer also works
       dispatch(apiClientInitSucceeded({ networkId, networkName: network.name }))
     } catch (e) {
+      // Discuss: Do we want to capture client init errors?
```
[the remark this answers is not in the document]
Comment (mvaivre) [reply src/App.tsx:134]: Agreed.

File: src/contexts/walletconnect.tsx (hunk 3 above)
Comment (mvaivre) [reply src/contexts/walletconnect.tsx]: So let's start without it. As a first step, let's start with the number of successful WalletConnect sessions only, very interesting metric.

File: src/components/Buttons/ShortcutButtons.tsx [hunk 5]
```diff
@@ -73,31 +73,31 @@ const ShortcutButtons = ({
   const lockWallet = () => {
     dispatch(walletLocked())
 
-    posthog?.capture('Locked wallet', { origin: analyticsOrigin })
+    posthog.capture('Locked wallet', { origin: analyticsOrigin })
```
Comment (mvaivre) [inline src/components/Buttons/ShortcutButtons.tsx:76]: Need to check, but isn't there another "Locked wallet" event somewhere, not necessarily attached to the shortcut button? 

What would be interesting here I believe, is to identify which shortcut button is clicked. Ie. "is it the one on the home page"? 

Same for the events below. WDYT?

**Review by mvaivre — APPROVED** [review]
Great job!

File: src/components/Buttons/ShortcutButtons.tsx (hunk 5 above)
Comment (nop33) [reply src/components/Buttons/ShortcutButtons.tsx:76]: This is already the case :) 

image

**Pull request merged**